pkg/scheduler/backend/queue/active_queue.go

```diff
@@ -323,12 +323,13 @@ func (aq *activeQueue) unlockedPop(logger klog.Logger) (*framework.QueuedPodInfo
 	}
 	aq.schedCycle++
 
-	// Update metrics and reset the set of unschedulable plugins for the next attempt.
+	// Update metrics for unschedulable plugins.
+	// Note: We don't clear UnschedulablePlugins and PendingPlugins here because:
+	// 1. If the pod schedules successfully, we need them for logging/debugging
+	// 2. If the pod fails to schedule, they will be cleared and repopulated in handleSchedulingFailure
 	for plugin := range pInfo.UnschedulablePlugins.Union(pInfo.PendingPlugins) {
 		metrics.UnschedulableReason(plugin, pInfo.Pod.Spec.SchedulerName).Dec()
 	}
-	pInfo.UnschedulablePlugins.Clear()
-	pInfo.PendingPlugins.Clear()
 	pInfo.GatingPlugin = ""
 	pInfo.GatingPluginEvents = nil
```

Comment:
nit: I know GatingPlugin and GatingPluginEvents can still be cleared here as well, but I prefer to clean up all these similar fields at the same place. So, let's create a new method on pInfo, something like `ClearRejectorPlugins()` and clear all of those four fields at `handleSchedulingFailure`?